docs/addons/vpc-cni.md

```diff
@@ -8,43 +8,124 @@ Installing VPC CNI as [Amazon EKS add-on](https://docs.aws.amazon.com/eks/latest
 
+Amazon EKS VPC CNI Addon now supports advanced configurations which means we can now pass configuration values as environment variables for setting up advanced configurations in Amazon VPC CNI. Please refer [Amazon EKS add-ons: Advanced configuration](https://aws.amazon.com/blogs/containers/amazon-eks-add-ons-advanced-configuration/) for more informatoion.
+
 ## Prerequisite
 - Amazon EKS add-ons are only available with Amazon EKS clusters running Kubernetes version 1.18 and later.
 
 ## Usage
 
+This add-on can used with two different patterns :
+
+Pattern # 1 : Simple and Easy. With all default values. This pattern wont create customer networking or setup any environment variables as part of Configuration Values.
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.VpcCniAddOn();
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .build(app, 'my-stack-name');
+```
+
+Pattern # 2 : Custom networking with new Secondary CIDR ranges. This pattern will first create Secondary CIDRs and Secondary Subnets with specified range of CIDRs as shown below in `resourceProvider` command. Then the VPC CNI addon will setup custom networking based on the parameters `awsVpcK8sCniCustomNetworkCfg`, `eniConfigLabelDef: "topology.kubernetes.io/zone"` for your Amazon EKS cluster workloads with created secondary subnet ranges to solve IP exhaustion.
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.VpcCniAddOn({
+  customNetworkingConfig: {
+      subnets: [
+          blueprints.getNamedResource("blueprint-construct-secondary-subnet0"),
+          blueprints.getNamedResource("blueprint-construct-secondary-subnet1"),
+          blueprints.getNamedResource("blueprint-construct-secondary-subnet2"),
+      ]   
+  },
+  awsVpcK8sCniCustomNetworkCfg: true,
+  eniConfigLabelDef: 'topology.kubernetes.io/zone'
+});
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .resourceProvider(blueprints.GlobalResources.Vpc, new VpcProvider(undefined,"10.64.0.0/24",["10.64.0.0/25","10.64.0.128/26","10.64.0.192/26"],))
+  .build(app, 'my-stack-name');
+```
+
+Pattern # 3 : Custom networking with custom VPC and Secondary Subnets. This pattern will use the custom VPC ID and Secondary subnet IDs passed by the user to create the blueprints stack. Then the VPC CNI addon will setup custom networking based on the parameters `awsVpcK8sCniCustomNetworkCfg`, `eniConfigLabelDef: "topology.kubernetes.io/zone"` for your Amazon EKS cluster workloads with passed secondary subnet ranges to solve IP exhaustion. 
```

Comment:
This pattern is unclear for the customers to use. We should tell them: register your secondary subnets under any arbitrary name as a resource provider (link) , like this

```typescript
...builder()
  .resourceProvider("secondary-subnet-1", new LookupSubnetProvider(subnet1Id)
  .resourceProvider("secondary-subnet-2", new LookupSubnetProvider(subnet2Id)
  .addOns(new blueprints.addons.VpcCniAddOn({
  customNetworkingConfig: {
      subnets: [
          blueprints.getNamedResource("secondary-subnet-1")
          ...
```

Reply:
Done